### ISSUE TITLE
Support docker export on Windows

### ISSUE THREAD

**Issue by dhirschfeld** [moby/moby#33581]
Exporting a container on Windows seems to be explicitly disallowed:
```go
if runtime.GOOS == "windows" {
    return fmt.Errorf("the daemon on this platform does not support export of a container")
}
````
https://github.com/moby/moby/blob/master/daemon/export.go#L16-L18

It would be good if this functionality were also available to Windows users.

**Comment by fasihrana** [comment]
Sorely need this. My images are 26GB in size!

**Comment by bartverstrynge** [comment]
+1

**Comment by dhirschfeld** [comment]
It's a bit of a perfect storm because you can't export your image and squash doesn't work - #31468.

If this isn't likely to be supported in the near future it would be great if squash could be made to work on Windows.

**Comment by friism** [comment]
I know it may not be a perfect replacement for all use cases, but multi-stage builds _do_ work on Windows and can achieve some of the same effects: https://docs.docker.com/engine/userguide/eng-image/multistage-build/

**Comment by dhirschfeld** [comment]
Yep, I'm using them and the do alleviate some of the image bloat issue.

**Comment by alexivkin** [comment]
@friism multistage only works for contained cases. consider this:
COPY HugeInstaller.exe c:\
RUN c:\HugeInstaller.exe; rm c:\HugeInstaller.exe; 
is a perfect case for when squash works and multi-stage does not. That windows installer can be spraying changes all over the filesystem and registry hives, making the multistage COPY very big and fragile

**Comment by mitchbte** [comment]
Is there any news on support for this feature??
Definitely keen to reduce the size of some images...
Come on guys it's an environmentally "green" feature, it should be at the top of the list. ;)

**Comment by judavi** [comment]
@jhowardmsft please could you mentions some of limitations?  I have been searching about the windowsfilter graphdriver but I haven't found good documentation. Will be interesting get an idea what are the blockers to make this works.

**Comment by jhowardmsft** [comment]
It's been a long while since I looked at it, but IIRC, both this and squash rely on diffs between arbitrary layers which the platform itself can't support with the current implementation of the filterdriver in Windows itself.

**Comment by renaudl** [comment]
that's a shame, first thing I tried on my first windows container.

**Comment by sdg002** [comment]
Same problem when I tried using **docker export** command
"the daemon on this operating system does not support exporting Windows containers"

Is there an alternative? I am using docker image with MSSQL server and if I am not able to save state then I am not sure if there is anything practical I can do.

**Comment by ohault** [comment]
+1

**Comment by hrvojesuta** [comment]
+1
